### PR TITLE
Mailrelay private dns zone

### PR DESCRIPTION
**Pull request by hannah38**
https://tools.hmcts.net/jira/browse/DTSPO-2940
Linked to cft ithc for testing as per ticket
A records with F5 LB IPs
Cname currently pointed at uks A records, but can be switched when needed.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: environments/prod/mailrelay-platform-hmcts-internal.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: "mailrelay.platform.hmcts.internal"
```
Comment (timja) [inline environments/prod/mailrelay-platform-hmcts-internal.yml:1]: why are we using an invalid domain? and also one we don't control?

if you want a separate domain something like:

internal.platform.hmcts.net

or 

platform-int.hmcts.net (would require getting control of)

but before you do anything can you explain the rationale?

**Review by timja — CHANGES_REQUESTED** [review]

File: environments/prod/mailrelay-platform-hmcts-internal.yml (hunk 1 above)
Comment (hannah38) [reply environments/prod/mailrelay-platform-hmcts-internal.yml:1]: Hey, I totally agree with you here and I'm updating it. We did talk about this in refinement and I have mailrelay.internal.hmcts.net in my notes - possibly the wrong one ended up on the ticket.

File: environments/prod/mailrelay-platform-hmcts-internal.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: "mailrelay.internal.platform.hmcts.net"
```
Comment (timja) [inline environments/prod/mailrelay-platform-hmcts-internal.yml:1]: also needs adding to https://github.com/hmcts/azure-private-dns/tree/master/components/prod

File: environments/prod/mailrelay-platform-hmcts-internal.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+vnet_links:
+  - link_name: "core-infra-vnet-ithc-internal"
+    vnet_id: "/subscriptions/7a4e3bd5-ae3a-4d0c-b441-2188fee3ff1c/resourceGroups/core-infra-ithc/providers/Microsoft.Network/virtualNetworks/core-infra-vnet-ithc"
+  - link_name: "core-infra-vnet-idam-ithc-internal"
+    vnet_id: "/subscriptions/7a4e3bd5-ae3a-4d0c-b441-2188fee3ff1c/resourceGroups/core-infra-idam-ithc/providers/Microsoft.Network/virtualNetworks/core-infra-vnet-idam-ithc"
+  - link_name: "mgmt-infra-prod"
```
Comment (timja) [inline environments/prod/mailrelay-platform-hmcts-internal.yml:7]: this isn't used for anything would be good to remove in a separate PR

File: environments/prod/mailrelay-platform-hmcts-internal.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    vnet_id: "/subscriptions/8999dec3-0104-4a27-94ee-6588559729d1/resourceGroups/mgmt-infra-prod/providers/Microsoft.Network/virtualNetworks/mgmt-infra-prod"
+  - link_name: "core-infra-vnet-mgmt"
+    vnet_id: "/subscriptions/ed302caf-ec27-4c64-a05e-85731c3ce90e/resourceGroups/rg-mgmt/providers/Microsoft.Network/virtualNetworks/core-infra-vnet-mgmt"
+  - link_name: "reform-mgmt-core-vnet"
+    vnet_id: "/subscriptions/ed302caf-ec27-4c64-a05e-85731c3ce90e/resourceGroups/reformMgmtCoreRG/providers/Microsoft.Network/virtualNetworks/reformMgmtCoreVNet"
+  - link_name: "bastion-prod-vnet"
+    vnet_id: "/subscriptions/2b1afc19-5ca9-4796-a56f-574a58670244/resourceGroups/bastion-prod-rg/providers/Microsoft.Network/virtualNetworks/bastion-prod-vnet"
+A:
+  - name: dev-in-uks
```
Comment (timja) [inline environments/prod/mailrelay-platform-hmcts-internal.yml:16]: what does `in` signify? if it's internal then that's in your new domain anyway

**Review by timja — CHANGES_REQUESTED** [review]

**Review by luigibk — APPROVED** [review]

File: environments/prod/mailrelay-platform-hmcts-internal.yml (hunk 4 above)
Comment (luigibk) [reply environments/prod/mailrelay-platform-hmcts-internal.yml]: "in" is not for internal, it's for inbound as there's an outbound towards the connector

File: environments/prod/mailrelay-platform-hmcts-internal.yml (hunk 1 above)
Comment (luigibk) [reply environments/prod/mailrelay-platform-hmcts-internal.yml]: Yeah good idea. All my fault :-) Thanks Tim

**Review by timja — APPROVED** [review]
